### DISCUSSION TITLE
Explore Dart Pub Workspaces for Serverpod

### DISCUSSION THREAD

**Discussion by rubenferreira97** [serverpod/serverpod#3056] [General]
Dart 3.6 introduced support for [pub workspaces (monorepo support)](https://dart.dev/tools/pub/workspaces). This new feature enables managing multiple packages within a single repository, simplifying dependency management and development workflows.

Given the similarities between the structure of Serverpod projects and monorepos, it seems like a natural fit to explore how Serverpod could leverage pub workspaces. Adopting this approach might provide advantages in organizing tightly coupled projects, like:

Improved dependency management across the project components.

Easier local development without the need for external dependency overrides.

Unified tooling and workflows for all packages within the workspace.

Would the Serverpod team be open to evaluating the feasibility of integrating or supporting pub workspaces in the Serverpod ecosystem? This could potentially enhance the developer experience and align with the latest advancements in the Dart ecosystem.

**Comment by Isakdl** [comment]
I think this makes a lot of sense, my only concern is the requirement on the dart version. I wouldn't want to require users to use dart 3.6 as the minimum version (yet).

**Reply by braj065** [reply]
Can it be optional/experimental for who are not using dart 3.6 yet?

**Reply by Isakdl** [reply]
Serverpod is fully compatible with workspaces! You can use it as long as your min dart version is set to 3.6.

**Reply by qiqetes** [reply]
I think should be part of the default template now. Or at least give an option to get it configured like this, shouldn't be too hard, I can provide a draft if you like.

**Reply by SandPod** [reply]
@qiqetes - We agree with you and are attempting to implement it soon, tracked in this issue: https://github.com/serverpod/serverpod/issues/3193
